include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1466,6 +1466,119 @@ struct __partial_merge_kernel
     }
 };
 
+//Searching for an intersection of virtual matrix (n1, n2) diagonal with the Merge Path to define sub-ranges
+//to serial merge. For example, a virtual matrix for [0,1,1,2,3] and [0,0,2,3] shown below:
+//     0   1  1  2   3
+//    ------------------
+//   |--->
+// 0 | 0 | 1  1  1   1
+//   |   |
+// 0 | 0 | 1  1  1   1
+//   |   ---------->
+// 2 | 0   0  0  0 | 1
+//   |             ---->
+// 3 | 0   0  0  0   0 |
+template<typename Rng1, typename Rng2, typename _Index, typename _Index1, typename _Index2>
+auto __find_start_point(Rng1& __rng1, Rng2& __rng2, _Index __i_elem, _Index1 __n_1, _Index2 __n_2)
+{
+    _Index1 __end1 = 0;
+    _Index2 __end2 = 0;
+    if (__i_elem < __n_1) //a condition to specify upper or down part of the merge matrix to be processed
+    {
+        auto __q = __i_elem; //digonal index
+        auto __n_diag = __q < __n_2 ? __q : __n_2; //digonal size
+
+        oneapi::dpl::counting_iterator<_Index> __diag_it(0);
+        auto __res = std::lower_bound(__diag_it, __diag_it + __n_diag, 1, [&__rng1, &__rng2, __q](auto __v1, auto __v2)
+            {
+                auto __val1 = __rng1[__q - __v1 - 1] < __rng2[__v1];
```

Comment:
As far as I see, it is possible to get integer an integer underflow when `__q` and `v1` are `0`.